test_communication/CMakeLists.txt

```diff
@@ -108,6 +108,38 @@ if(BUILD_TESTING)
       "rclcpp${target_suffix}")
   endfunction()
 
+  function(custom_test_c target)
+    # get typesupport of rmw implementation to include / link against the corresponding interfaces
+    get_rmw_typesupport(typesupport_impls "${rmw_implementation}" LANGUAGE "c")
+
+    ament_add_gtest(
+      "${target}${target_suffix}" ${ARGN}
+      TIMEOUT 90
+      ENV RCL_ASSERT_RMW_ID_MATCHES=${rmw_implementation})
+    target_link_libraries(${target}${target_suffix}
+      ${_AMENT_EXPORT_ABSOLUTE_LIBRARIES}
+      ${_AMENT_EXPORT_LIBRARY_TARGETS})
+    add_dependencies(${target}${target_suffix} ${PROJECT_NAME})
+    foreach(typesupport_impl ${typesupport_impls})
+      rosidl_target_interfaces(${target}${target_suffix}
+        ${PROJECT_NAME} ${typesupport_impl})
+    endforeach()
+    rosidl_target_interfaces(${target}${target_suffix}
+      ${PROJECT_NAME} rosidl_generator_c)
+    ament_target_dependencies(${target}${target_suffix}
+      "rcl${target_suffix}")
+    # register executable as a test
```

Comment:
artefact indeed, removed in https://github.com/ros2/system_tests/pull/122/commits/09b6a18f765d78ac8e2035c982df9135a109b725